lib/buffer.js

```diff
@@ -151,30 +150,20 @@ function fromObject(obj) {
   }
 
   if (obj.buffer instanceof ArrayBuffer || obj.length) {
```

Comment:
My bad. We are missing a corner case here. If the length of the array is zero we will throw an error. Perhaps we can change this to `'length' in obj`.